src/libsyntax/parse/parser.rs

```diff
@@ -4437,6 +4438,36 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_trait_item_assoc_ty(&mut self, preceding_attrs: Vec<Attribute>)
+        -> PResult<'a, (ast::Generics, TyParam)> {
+        let span = self.span;
+        let ident = self.parse_ident()?;
+        let mut generics = self.parse_generics()?;
+
+        // Parse optional colon and param bounds.
+        let bounds = if self.eat(&token::Colon) {
+            self.parse_ty_param_bounds()?
+        } else {
+            Vec::new()
+        };
+
+        let default = if self.eat(&token::Eq) {
+            Some(self.parse_ty()?)
+        } else {
+            None
+        };
+        generics.where_clause = self.parse_where_clause()?;
```

Comment:
Again, the `where` clause is in unexpected position
```rust
// As implemented here
type AssociatedAlias<'a, T>: Trait + 'static = u8 where T: Copy;
```
while I would expect it to be
```rust
type AssociatedAlias<'a, T>: Trait + 'static where T: Copy = u8;
```
, i.e. consistent with existing type aliases and traits
```rust
type UsualAlias<'a, T> where T: Copy = u8;
type MyTrait<'a, T>: Trait + 'static where T: Copy {}
```

Reply:
Thanks! I was wondering about this too. I'll correct it. :smile: